flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactoryOptions.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ConfigOptions;
+
+public class FlinkCatalogFactoryOptions {
+  private FlinkCatalogFactoryOptions() {
+  }
+
+  public static final String IDENTIFIER = "iceberg";
+
+  public static final ConfigOption<String> CATALOG_TYPE =
+      ConfigOptions.key("catalog-type")
+          .stringType()
+          .noDefaultValue()
+          .withDescription("Iceberg catalog type, 'hive' or 'hadoop'");
+
+  public static final ConfigOption<String> PROPERTY_VERSION =
+      ConfigOptions.key("property-version")
+          .stringType()
+          .defaultValue("1")
+          .withDescription(
+              "Version number to describe the property version. This property can be used for backwards " +
+                  "compatibility in case the property format changes. The current property version is `1`. (Optional)");
+
+  public static final ConfigOption<String> URI =
```

Comment:
can we reuse some constants from `CatalogProperties` in iceberg-core module? like
```
ConfigOptions.key(CatalogProperties.URI)
```